langchain_tools_demo/tools.py

```diff
@@ -0,0 +1,256 @@
+    )
+    if response.status_code != 200:
+        return f"Error trying to find flight: {response.text}"
+
+    return response.json()
+
+
+class ListFlights(BaseModel):
+    departure_airport: Optional[str] = Field(
+        description="Departure airport 3-letter code",
+    )
+    arrival_airport: Optional[str] = Field(description="Arrival airport 3-letter code")
+    date: Optional[str] = Field(description="Date of flight departure")
+
+
+@tool("List Flights", args_schema=ListFlights)
+def list_flights(departure_airport: str, arrival_airport: str, date: str):
+    """
+    Use this tool to list all flights matching search criteria.
+    Takes an arrival airport and departure airport and returns all flights.
+    The agent can decide to return the results directly to the user.
+    Input of this tool must be in JSON format and include all three inputs - arrival_airport, departure_airport, and date.
+    Example:
+    {{
+        "departure_airport": "SFO",
+        "arrival_airport": null,
+        "date": null
+    }}
+    Example:
+    {{
+        "departure_airport": "SFO",
+        "arrival_airport": "SEA",
+        "date": "2023-11-01"
+    }}
+    Example:
+    {{
+        "departure_airport": null,
+        "arrival_airport": "SFO",
+        "date": "2023-01-01"
+    }}
+    """
+    response = get_request(
+        f"{BASE_URL}/flights/search",
+        {
+            "departure_airport": departure_airport,
+            "arrival_airport": arrival_airport,
+            "date": date,
+        },
+    )
+    if response.status_code != 200:
+        return f"Error searching flights: {response.text}"
+
+    num = 2
+    if len(response.json()) < 1:
+        return "There are no flights matching that query. Let the user know there are no results."
+    elif len(response.json()) > num:
+        return (
+            f"There are {len(response.json())} flights matching that query. Here are the first {num} results:\n"
+            + " ".join([f"{response.json()[i]}" for i in range(num)])
+        )
+    else:
+        return "\n".join([f"{r}" for r in response.json()])
+
+
+# Amenities
+class AmenityIdInput(BaseModel):
+    id: int = Field(description="Unique identifier")
+
+
+@tool("Get Amenity", args_schema=AmenityIdInput)
+def get_amenity(id: int):
+    """
+    Use this tool to get info for a specific airport amenity.
+    Takes an id and returns info on the amenity.
+    Do NOT guess an amenity id. Use Search Amenities to search by name.
+    Always use the id from the search_amenities tool.
+    """
+    response = get_request(
+        f"{BASE_URL}/amenities",
+        {"id": id},
+    )
+    if response.status_code != 200:
+        return f"Error trying to find amenity: {response.text}"
+
+    return response.json()
+
+
+class QueryInput(BaseModel):
+    query: str = Field(description="Search query")
+
+
+@tool("Search Amenities", args_schema=QueryInput)
+def search_amenities(query: str):
```

Comment:
Maybe we should call this tool `list_amenities` to match `list_flights`?

Reply:
I like search amenities because it's using the vector similarity search, but I will leave this open for future discussion